lib/instream/log/default_logger.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Instream.Log.DefaultLogger do
+  @moduledoc """
+  Default logger for all entries.
+  """
+
+  require Logger
+
+  alias Instream.Log.PingEntry
+  alias Instream.Log.QueryEntry
+  alias Instream.Log.StatusEntry
+  alias Instream.Log.WriteEntry
+
+  @doc """
+  Logs a request.
+  """
+  @spec log(Instream.Connection.log_entry) :: Instream.Connection.log_entry
+  def log(%PingEntry{} = entry) do
+    Logger.debug(fn ->
+      [ "[ping ", entry.host, "] ", to_string(entry.result) ]
+    end, metadata(entry))
+
+    entry
+  end
+
+  def log(%QueryEntry{} = entry) do
+    Logger.debug(fn ->
+      [ "[query] ", entry.query ]
+    end, metadata(entry))
+
+    entry
+  end
+
+  def log(%StatusEntry{} = entry) do
+    Logger.debug(fn ->
+      [ "[status ", entry.host, "] ", to_string(entry.result) ]
+    end, metadata(entry))
+
+    entry
+  end
+
+  def log(%WriteEntry{} = entry) do
+    Logger.debug(fn ->
+      [ "[write] ", to_string(entry.points), " points" ]
+    end, metadata(entry))
+
+    entry
+  end
+
+
+  # Utility methods
+
+  defp metadata(%{ metadata: metadata }) do
+    metadata
+    |> Map.delete(:__struct__)
+    |> Keyword.new()
```

Comment:
In this case not as we are handling a struct (`Instream.Log.Metadata`) and not a plain map:

```shell
iex(1)> Instream.Log.Metadata |> Enum.into([])
** (Protocol.UndefinedError) protocol Enumerable not implemented for Instream.Log.Metadata
```